dotCMS/src/main/java/com/dotmarketing/util/importer/ImportResultConverter.java

```diff
@@ -120,6 +128,52 @@ public static void addValidationResultsToLegacyMap(
         }
     }
 
+    /**
+     * Converts the structured LineImportResult to the legacy HashMap format. This method ensures
+     * backward compatibility by formatting the structured data into the string-based format
+     * expected by legacy code.
+     *
+     * @param importResults The structured LineImportResult to convert
+     * @param legacyResults The legacy format results map to update
+     * @param counters      The import counters to update
+     */
+    public static void lineImportResultToLegacyMap(
+            final LineImportResult importResults,
+            final Map<String, List<String>> legacyResults,
+            final Counters counters) {
+
+        // Convert messages
+        for (ValidationMessage msg : importResults.messages()) {
+            switch (msg.type()) {
+                case ERROR:
+                    legacyResults.get("errors").add(formatMessage(msg));
+                    break;
+                case WARNING:
+                    legacyResults.get("warnings").add(formatMessage(msg));
+                    break;
+                case INFO:
+                    legacyResults.get("messages").add(formatMessage(msg));
+                    break;
```

Comment:
perhaps we can have these strings ("messages","warnings","errors") as constants or part of the Enum that serves as the switch case

Reply:
Same with the other string "lastInode"

Reply:
ImportResultConverter needs a private constructor to guarantee the class cannot be instantiated as it only has static methods

Reply:
Done